libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -1,57 +1,212 @@
+    Bolt11(LNInvoice invoice);
+    NodeId(string node_id);
+    Url(string url);
+    LnUrlPay(LnUrlPayRequestData data);
+    LnUrlWithdraw(LnUrlWithdrawRequestData data);
+    LnUrlAuth(LnUrlAuthRequestData data);
+    LnUrlEndpointError(LnUrlErrorData data);
+};
+
+dictionary BitcoinAddressData {
+    string address;
+    Network network;
+    u64? amount_sat;
+    string? label;
+    string? message;
+};
+
+dictionary LnUrlPayRequestData {
+    string callback;
+    u64 min_sendable;
+    u64 max_sendable;
+    string metadata_str;
+    u16 comment_allowed;
+    string domain;
+    boolean allows_nostr;
+    string? nostr_pubkey;
+    string? ln_address;
+};
+
+dictionary LnUrlAuthRequestData {
+    string k1;
+    string domain;
+    string url;
+    string? action = null;
+};
+
+dictionary LnUrlErrorData {
+    string reason;
+};
+
+[Enum]
+interface SuccessActionProcessed {
+    Aes(AesSuccessActionDataResult result);
+    Message(MessageSuccessActionData data);
+    Url(UrlSuccessActionData data);
+};
+
+[Enum]
+interface AesSuccessActionDataResult {
+    Decrypted(AesSuccessActionDataDecrypted data);
+    ErrorStatus(string reason);
+};
+
+dictionary AesSuccessActionDataDecrypted {
+    string description;
+    string plaintext;
+};
+
+dictionary MessageSuccessActionData {
+    string message;
+};
+
+dictionary UrlSuccessActionData {
+    string description;
+    string url;
+};
+
+dictionary LnUrlPayErrorData {
+    string payment_hash;
+    string reason;
+};
+
+dictionary LnUrlPayRequest {
+    LnUrlPayRequestData data;
+    u64 amount_msat;
+    string? comment = null;
+    string? payment_label = null;
+};
 
 [Error]
-enum SdkError {
-    "Generic",
-    "ServiceConnectivity",
+interface LnUrlPayError {
+    AlreadyPaid();
+    Generic(string err);
+    InvalidAmount(string err);
+    InvalidInvoice(string err);
+    InvalidNetwork(string err);
+    InvalidUri(string err);
+    InvoiceExpired(string err);
+    PaymentFailed(string err);
+    PaymentTimeout(string err);
+    RouteNotFound(string err);
+    RouteTooExpensive(string err);
+    ServiceConnectivity(string err);
+};
+
+dictionary MetadataItem {
+    string key;
+    string value;
+};
+
+dictionary LnUrlWithdrawRequest {
+    LnUrlWithdrawRequestData data;
+    u64 amount_msat;
+    string? description = null;
+};
+
+dictionary LnUrlWithdrawRequestData {
+    string callback;
+    string k1;
+    string default_description;
+    u64 min_withdrawable;
+    u64 max_withdrawable;
 };
 
 [Error]
-enum ConnectError {
+enum LnUrlWithdrawError {
     "Generic",
-    "RestoreOnly",
+    "InvalidAmount",
+    "InvalidInvoice",
+    "InvalidUri",
     "ServiceConnectivity",
+    "InvoiceNoRoutingHints",
 };
 
+// END sdk-common mirror imports
+////////////////////////////////
+
+
+//////////////////////////////////
+// BEGIN sdk-common wrappers
+// These are connecting structures that glue relevant sdk-common structs to the SDK
+
+[Enum]
+interface WrappedLnUrlPayResult {
```

Comment:
> Any reason why we call it `WrappedLnUrlPayResult`?
Because it partially overlaps, partially adds extra data to the `sdk-common::LnUrlPayResult`.

The discussed approach to stick to the `sdk-common` returned structs, and duplicate them in the SDK where necessary, unfortunately didn't work so I needed a different approach.

The `sdk-common` LNURL-pay methods directly return a

```rust
pub enum LnUrlPayResult {
    EndpointSuccess { data: LnUrlPaySuccessData },
    EndpointError { data: LnUrlErrorData },
    PayError { data: LnUrlPayErrorData },
}
```

but in case of success, the SDK has to include the `Payment`, so the SDK method returns

```rust
 // Wrapped.. because inner data below
pub enum WrappedLnUrlPayResult {
    // Inner data = sdk-common struct + payment
    EndpointSuccess { data: WrappedLnUrlPaySuccessData },

    // Rest is identical
    EndpointError { data: LnUrlErrorData },
    PayError { data: LnUrlPayErrorData },
}
```

>  I think it should be as before, no need for changing the naming.

One way to achieve that would be
- rename `Wrapped..` into the `sdk-common` names (so to the SDK user they're unchanged)
- rename the `sdk-common` names into `Base..` or `Raw..`

What do you think?

---

> is there any other changes to the udl? If not can we revert to the original udl file because as it is currently it is difficult to detect the changes.

Done, I moved segments around to simplify the diff for easier review: 0df7665685bea6787980111d48af8179098e29fd

I would still move them back after review, to make the UDL more structured:
- `sdk-common` definitions
- "glue" or wrapper definitions
- SDK definitions

Reply:
I think we just need to remove the Wrapped prefix and stay with the old names. I see no problem that both sdk-common and sdk-core use the same names as they exist in different modules and creates.

Reply:
Done: b1b249129eda9a80231ae3af8101700db5ade97d